#**1.Quantum Information Channels**

Quantum information channels are information transmission systems based on the principles of quantum mechanics. Unlike traditional information transmission systems, quantum information channels transmit and process quantum states.

A quantum information channel is typically designed as a physical system that transmits a quantum state. Quantum states, replacing the classical bits used in classical information transfer, are called quantum bits or "qubits." Unlike classical bits, qubits have the ability to exist in both 0 and 1 simultaneously, enabling parallel processing. This potential for parallelism can lead to increased speed and efficiency in quantum information processing and transmission.

Quantum information channels can utilize various quantum mechanical phenomena to enable the transmission, processing, and reception of quantum states. Among these phenomena are photonic quantum states for quantum telecommunication (e.g., quantum key distribution), nuclear magnetism for quantum information processing (e.g., nuclear magnetic resonance), and concepts like quantum entanglement for quantum entanglement.

Quantum information channels have significant applications in areas such as quantum information processing, quantum communication, and quantum computing. They are particularly used to develop secure and fast communication protocols such as quantum encryption, quantum key distribution, and quantum teleportation.



Within the scope of this topic, Qiskit hosts the **qiskit.quantum_info** module. This module contains several classes representing quantum information channels. Below is a table listing the classes representing quantum information channels in the **qiskit.quantum_info module** along with their descriptions:

| Class Name   | Description                                                                   |
|--------------|-------------------------------------------------------------------------------|
| Choi         | Representation of the Choi matrix of a quantum channel.                       |
| SuperOp      | Representation of the superoperator of a quantum channel.                     |
| Kraus        | Representation of the Kraus operators of a quantum channel.                   |
| Stinespring  | Representation of the Stinespring operator of a quantum channel.              |
| Chi          | Representation of the Pauli basis Chi matrix of a quantum channel.            |
| PTM          | Representation of the Pauli Transfer Matrix (PTM) of a quantum channel.       |

Let's discuss an example right away.

Here we will use the *Kraus* representation.

In [23]:
#1.
from qiskit import QuantumCircuit,transpile
from qiskit.quantum_info import Kraus
from qiskit_aer import Aer
import numpy as np

In [24]:
#2.
noise_ops = [np.sqrt(0.9) * np.array([[1, 0],[0, 1]]), np.sqrt(0.1) * np.array([[0, 1], [1, 0]])]
kraus = Kraus(noise_ops)

**We may need to explain this part.**

* This code defines noise operators in the context of Quantum Error Correction (QEC) or Quantum Noise modeling.

**Now let's elaborate a bit more.**

* This code defines noise operators in the context of Quantum Error Correction (QEC) or Quantum Noise modeling.

  * First, a list named `noise_ops` is defined. Each item in this list represents a noise operator. The expression `np.sqrt(0.9) * np.array([[1, 0],[0, 1]])` matches with the scaled version of the identity matrix by √0.9. This represents a "low-density" error in a system, meaning the identity operator remains unchanged.

  * The second item is defined with the expression `np.sqrt(0.1) * np.array([[0, 1], [1, 0]])`, representing a bit flip error. Here, a Pauli X (bit flip) operator scaled by √0.1 is used. This type of error alters the state of a qubit. For example, a qubit in the state 0 can flip to 1 due to an error, or vice versa.

  * Finally, an object from the `Kraus` class is instantiated, and the `noise_ops` list is passed as a parameter to it. The `Kraus` class is a commonly used structure in a QEC model or noise modeling. Kraus operators are matrices representing possible error types on a qubit. These errors can affect the superposition states and correlations of qubits.

This code snippet creates a model containing two types of operators representing noise (identity and bit flip). This model can then be used in designing quantum error correction algorithms or other techniques aimed at mitigating noise.

In [25]:
#3.
qc = QuantumCircuit(2)
qc.append(kraus, [0])
qc.append(kraus, [1])

In [26]:
#4.
qc.measure_all()
qc.draw()

┌───────┐ ░ ┌─┐   
   q_0: ┤ kraus ├─░─┤M├───
        ├───────┤ ░ └╥┘┌─┐
   q_1: ┤ kraus ├─░──╫─┤M├
        └───────┘ ░  ║ └╥┘
meas: 2/═════════════╩══╩═
                     0  1

In [27]:
#5.
backend = Aer.get_backend("aer_simulator")
tqc = transpile(qc, backend)
job = backend.run(tqc, shots=1000)
result = job.result()
counts = result.get_counts(tqc)
print(counts)

{'11': 6, '01': 90, '00': 825, '10': 79}


Now let's discuss the output here.

1. This output represents the results obtained from simulating a 2-qubit quantum circuit.

2. These results show the statistical outcomes obtained after 1000 shots on a specific 2-qubit configuration in a quantum circuit. Each outcome represents a specific qubit configuration.

**For example:**
- 11 occurrences were obtained in 6 shots.
- 01 occurrences were obtained in 90 shots.
- 10 occurrences were obtained in 79 shots.
- 00 occurrences were obtained in 825 shots.